helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithmFactory.java

```diff
@@ -19,19 +19,34 @@
  * under the License.
  */
 
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 
+import org.apache.helix.HelixManagerProperties;
+import org.apache.helix.SystemPropertyKeys;
 import org.apache.helix.controller.rebalancer.waged.RebalanceAlgorithm;
 import org.apache.helix.model.ClusterConfig;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 
 /**
  * The factory class to create an instance of {@link ConstraintBasedAlgorithm}
  */
 public class ConstraintBasedAlgorithmFactory {
+  private static final String PARTITION_MOVEMENT_CONSTRAINT = "PartitionMovementConstraint";
+  private static final Map<String, Float> MODEL;
+
+  static {
+    MODEL = new HashMap<>();
```

Comment:
Let's have a default value set in case there are something wrong with the property file.